app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java

```diff
@@ -388,38 +392,19 @@ protected void onCreate(Bundle savedInstanceState) {
             }
         };
 
+        widthScalerLayout = binding.widthScalerLayout;
+        heightScalerLayout = binding.heightScalerLayout;
         centerGuideline = binding.centerGuideline;
-        barcodeScaler = binding.barcodeScaler;
-        barcodeScaler.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
-            @Override
-            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
-                if (!fromUser) {
-                    Log.d(TAG, "non user triggered onProgressChanged, ignoring, progress is " + progress);
-                    return;
-                }
-                Log.d(TAG, "Progress is " + progress);
-                Log.d(TAG, "Max is " + barcodeScaler.getMax());
-                float scale = (float) progress / (float) barcodeScaler.getMax();
-                Log.d(TAG, "Scaling to " + scale);
+        barcodeHeightScaler = binding.barcodeHeightScaler;
 
-                loyaltyCard.zoomLevel = progress;
-                DBHelper.updateLoyaltyCardZoomLevel(database, loyaltyCardId, loyaltyCard.zoomLevel);
+        SeekBarListener heightScalerListener = new SeekBarListener(barcodeHeightScaler);
+        barcodeHeightScaler.setOnSeekBarChangeListener(heightScalerListener);
 
-                setCenterGuideline(loyaltyCard.zoomLevel);
-
-                drawMainImage(mainImageIndex, true, isFullscreen);
-            }
-
-            @Override
-            public void onStartTrackingTouch(SeekBar seekBar) {
-
-            }
-
-            @Override
-            public void onStopTrackingTouch(SeekBar seekBar) {
-
-            }
-        });
+        // set zoom width of barcode
+        barcodeWidthScaler = binding.barcodeWidthScaler;
+        zoomWidthText = binding.zoomWidthText;
+        SeekBarListener widthScalerListener = new SeekBarListener(barcodeWidthScaler);
```

Comment:
Again inconsistent ordering:

Width
Height
Height
Width

And only the width has a comment and the height doesn't

Reply:
Also the extra `barcodeXScaler = binding.barcodeXScaler` seems unnecessary and `zoomWidthText` seems completely unused.